plugin/pkg/scheduler/generic_scheduler.go

```diff
@@ -125,6 +126,10 @@ func findNodesThatFit(pod *api.Pod, podLister algorithm.PodLister, predicateFunc
 				if _, found := failedPredicateMap[node.Name]; !found {
 					failedPredicateMap[node.Name] = util.StringSet{}
 				}
+				if predicates.FailedResourceType != "" {
+					failedPredicateMap[node.Name].Insert(predicates.FailedResourceType)
```

Comment:
I see. Using a global (package-level) variable is a little janky but we need to clean up the whole scheduler and can take care of it then.